src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1984,10 +2119,22 @@ internal static uint GetValEscape(BoundExpression expr, uint scopeOfTheContainin
 
                     return Math.Max(GetValEscape(binary.Left, scopeOfTheContainingExpression),
                                     GetValEscape(binary.Right, scopeOfTheContainingExpression));
-            }
 
-            // At this point we should have covered all the possible cases for anything that may return its operands or manufacture local references.
-            return Binder.ExternalScope;
+                case BoundKind.UserDefinedConditionalLogicalOperator:
+                    var uo = (BoundUserDefinedConditionalLogicalOperator)expr;
+
+                    return Math.Max(GetValEscape(uo.Left, scopeOfTheContainingExpression),
+                                    GetValEscape(uo.Right, scopeOfTheContainingExpression));
+
+                case BoundKind.QueryClause:
```

Comment:
deferring to ".Value" is what we do in lowering, so it makes sense to do the same in analysis.

I could not hit this with tests though. I keep running into limitations where ref-like types cannot implement `IEnumerable<T>` and cannot be the `T`